presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -113,6 +113,23 @@ public static Slice stAsText(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return Slices.utf8Slice(deserialize(input).asText());
     }
 
+    @SqlNullable
+    @Description("Returns a geometry that represents all points whose distance from specified geometry is less than or equal to specified distance")
+    @ScalarFunction("ST_Buffer")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stBuffer(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(StandardTypes.DOUBLE) double distance)
+    {
+        if (distance < 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Distance parameter for ST_Buffer must be greated then or equal to 0");
+        }
+
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.isEmpty()) {
+            return null;
+        }
+        return serialize(geometry.buffer(distance));
```

Comment:
When `distance` is `0` the `buffer` method returns `this`, which is the `geometry` instance here. Do you think it is worth optimizing that case in this method to get rid of the deser/ser overhead:

```
if (distance == 0) {
    return input;
}
```